addons/mass_mailing/static/src/js/mass_mailing_full_width_form.js

```diff
@@ -0,0 +1,208 @@
+        const windowHeight = $(window).height();
+        const $iframeDocument = this.$iframe.contents();
+        const $sidebar = $iframeDocument.find('#oe_snippets');
+        const isFullscreen =  this._isFullScreen();
+        if (isFullscreen) {
+            $sidebar.height(windowHeight);
+            this.$iframe.height(windowHeight);
+            $sidebar.css({
+                top: '',
+                bottom: '',
+            });
+        } else {
+            const iframeTop = this.$iframe.offset().top;
+            $sidebar.css({
+                height: '',
+                top: Math.max(0, this.$('.o_content').offset().top - iframeTop),
+                bottom: this.$iframe.height() - windowHeight + iframeTop,
+            });
+        }
+    },
+    /**
+     * Return true if the form is in full screen mode, false otherwise.
+     *
+     * @private
+     * @returns {boolean}
+     */
+    _isFullScreen() {
+        return window.top.document.body.classList.contains('o_field_widgetTextHtml_fullscreen');
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Assume the iframe was updated on each dom_updated event.
+     *
+     * @private
+     */
+    _onDomUpdated() {
+        const data = { $iframe: this.$('iframe.wysiwyg_iframe, iframe.o_readonly') };
+        this._onIframeUpdated({ data });
+    },
+    /**
+     * Resize the given iframe so its height fits its contents and initialize a
+     * resize observer to resize on each size change in its contents.
+     * This also ensures the contents of the sidebar remain visible no matter
+     * how much we resize the iframe and scroll down.
+     *
+     * @private
+     * @param {JQuery} ev.data.$iframe
+     */
+    _onIframeUpdated(ev) {
+        const $iframe = ev.data.$iframe;
+        if (!$iframe.length || !$iframe.contents().length) {
+            return;
+        }
+        const $iframeDoc = $iframe.contents();
+        if ($iframe !== this.$iframe) {
+            $iframeDoc.find('body').on('click', '.o_fullscreen_btn', this._onToggleFullscreen.bind(this));
+            this.$('.o_content').on('scroll', this._repositionSidebar.bind(this));
+        }
+        this.$iframe = $iframe;
+        this._autoresizeIframe();
+        const iframeTarget = $iframeDoc.find('#iframe_target');
+        if (iframeTarget[0]) {
+            const isFullscreen = this._isFullScreen();
+            iframeTarget.css({
+                display: isFullscreen ? '' : 'flex',
+                'flex-direction': isFullscreen ? '' : 'column',
+            });
+            if (this._resizeObserver) {
+                this._resizeObserver.disconnect();
+            }
+            this._resizeObserver = new ResizeObserver(this._onResizeIframeContents.bind(this));
+            this._resizeObserver.observe(iframeTarget[0]);
```

Comment:
Isn't that also something you need to register only when your iframe reference changes, in your test here above?